embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/PreviewQueryBuilder.java

```diff
@@ -0,0 +1,41 @@
+package org.embulk.input.jdbc;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.regex.Pattern;
+
+public class PreviewQueryBuilder
+{
+    private static final String MSSQL_PRODUCT_NAME = "Microsoft SQL Server";
```

Comment:
It does not seem very reasonable to have MSSQL-specific logic in this generic `PreviewQueryBuilder`. You may want to have a MSSQL-specific query builder in `embulk-input-sqlserver`, and use it there?  Then it does not require `connection` as well.

Or, it may apply also for others, MySQL and PostgreSQL. Even if their implementation would be the same, you may want to have a query builder for MySQL in `embulk-input-mysql`, and another query builder for PostgrSQL in `embulk-input-postgresql`. The single responsibility principle is often more prioritized than DRY.

Reply:
> Why 50? Do you have any specific reason why 50 is good there? Or, shouldn't it be user-configurable?
Actually, I don't know which number is best, so just pick random 50 records for preview is enough. It's better to move it to configurable, but I don't think we need it because it's just for display some sample record in our console UI.